source/common/tls/context_impl.h

```diff
@@ -193,13 +193,19 @@ class ClientContextImpl : public ContextImpl, public Envoy::Ssl::ClientContext {
 
 enum class OcspStapleAction { Staple, NoStaple, Fail, ClientNotCapable };
 
-class ServerContextImpl : public ContextImpl, public Envoy::Ssl::ServerContext {
+class ServerContextImpl : public ContextImpl,
+                          public Envoy::Ssl::ServerContext,
+                          public Envoy::Ssl::ContextSelectionCallback,
+                          public std::enable_shared_from_this<ServerContextImpl> {
 public:
   ServerContextImpl(Stats::Scope& scope, const Envoy::Ssl::ServerContextConfig& config,
                     const std::vector<std::string>& server_names,
                     Server::Configuration::CommonFactoryContext& factory_context,
                     Ssl::ContextAdditionalInitFunc additional_init);
 
+  // Ssl::ContextSelectionCallback
+  const std::vector<Ssl::TlsContext>& getTlsContexts() const override { return tls_contexts_; };
```

Comment:
it will be used in the extension implementation, i.e. [TestTlsContextProvider](https://github.com/envoyproxy/envoy/blob/450d1b13196588ad13875d6b0e0a01efa13c466e/test/common/tls/tls_context_provider_test.cc#L114).
so that an extension implementation could get the existing TLS contexts.